src/index.js

```diff
@@ -249,13 +249,17 @@ Multiaddr.prototype.decapsulate = function decapsulate (addr) {
  * }
  */
 Multiaddr.prototype.getPeerId = function getPeerId () {
-  let b58str = this.stringTuples().filter((tuple) => {
-    if (tuple[0] === protocols.names['ipfs'].code) {
-      return true
-    }
-  })[0][1]
+  let b58str = null
+  try {
+    b58str = this.stringTuples().filter((tuple) => {
+      if (tuple[0] === protocols.names['ipfs'].code) {
+        return true
+      }
+    })[0][1]
+
+    bs58.decode(b58str)
+  } catch (e) {}
 
-  bs58.decode(b58str)
   return b58str
```

Comment:
I want the utility function to be more ergonomic, in the majority of cases you want to just `ma.getPeerId()` from a multiaddr and not worry about wrapping it in a try/catch, if the id was missing or invalid, you'll get a null. The intent here is not to return an invalid b58 string and I think this accomplishes it?